src/renderers/shaders/ShaderChunk/shadowmap_pars_fragment.glsl

```diff
@@ -1,12 +1,17 @@
 #ifdef USE_SHADOWMAP
 
+	uniform int isShadowCube[ MAX_SHADOWS ];
+	uniform samplerCube shadowCube[ MAX_SHADOWS ];
 	uniform sampler2D shadowMap[ MAX_SHADOWS ];
 	uniform vec2 shadowMapSize[ MAX_SHADOWS ];
 
 	uniform float shadowDarkness[ MAX_SHADOWS ];
 	uniform float shadowBias[ MAX_SHADOWS ];
 
+	uniform vec3 shadowLightPosition[ MAX_SHADOWS ];
+
 	varying vec4 vShadowCoord[ MAX_SHADOWS ];
```

Comment:
> So as it turns out, all the calculations can still be done in world space (which makes things much easier) as long as the point light's position is converted to world space in the vertex shader. Since I was previously not making use of the 'shadowMatrix' uniform for point lights, I can use that matrix to perform the conversion by populating it with the inverse of the view matrix. Does that sound reasonable?

Note sure: It's a high base cost for the matrix. Without it, we get 4 light positions for "free", right?


Also beware the number of varying vectors. See `MAX_VARYING_VECTORS` and `MAX_FRAGMENT_UNIFORM_VECTORS` on [webglstats.com](http://webglstats.com/), switching off anything but "Smartphone".

> As for the 'isShadowCube' uniform, into which other existing uniform would you prefer that be packed?

I would probably end up just using the sign of an unsigned value and add a comment...